cmd/seasonpackarr/main.go

```diff
@@ -251,16 +256,17 @@ func handleSeasonPack(w http.ResponseWriter, r *http.Request) {
 				packDirName := utils.FormatSeasonPackTitle(req.Name)
 
 				childPath := filepath.Join(child.t.SavePath, fileName)
-				packPath := filepath.Join(viper.GetString("PreImportPath"), packDirName, fileName)
+				packPath := filepath.Join(cfg.Config.PreImportPath, packDirName, fileName)
 
 				createHardlink(childPath, packPath)
 
-				http.Error(w, fmt.Sprintf("created hardlink of %q into folder %q\n",
+				http.Error(w, fmt.Sprintf("created hardlink of %q into %q\n",
 					childPath, packPath), res)
 				continue
 			}
 		}
 	} else {
```

Comment:
Only a minor change, but having code "return early" is a nice pattern and get rids of some if/else chaining.

I would change to

```go
if v, ok := mp.e[utils.GetFormattedTitle(requestrls.r)]; !ok {
    log.Info().Msgf("unique submission: %q", req.Name)
    http.Error(w, fmt.Sprintf("unique submission: %q\n", req.Name), 200)
}

for _, child := range v {
	if checkCandidates(&requestrls, &child) == 210 {
		log.Info().Msgf("release already exists in client: %q", req.Name)
		http.Error(w, fmt.Sprintf("release already exists in client: %q\n", req.Name), 210)
		return
	}
}

for _, child := range v {
	res := checkCandidates(&requestrls, &child)

	if res == 210 {
		log.Info().Msgf("release already exists in client: %q", req.Name)
		http.Error(w, fmt.Sprintf("release already exists in client: %q\n", req.Name), res)
		break
	}
	if res == 211 {
		log.Info().Msgf("not a season pack: %q", req.Name)
		http.Error(w, fmt.Sprintf("not a season pack: %q\n", req.Name), res)
		break
	}
	if res == 250 {
		m, err := req.getFiles(child.t.Hash)
		if err != nil {
			log.Error().Err(err).Msgf("failed to get files for %q", req.Name)
			continue
		}

		fileName := ""
		for _, v := range *m {
			fileName = v.Name
			break
		}

		packDirName := utils.FormatSeasonPackTitle(req.Name)

		childPath := filepath.Join(child.t.SavePath, fileName)
		packPath := filepath.Join(cfg.Config.PreImportPath, packDirName, fileName)

		createHardlink(childPath, packPath)

		http.Error(w, fmt.Sprintf("created hardlink of %q into %q\n",
					childPath, packPath), res)
		continue
	}
}
```

And further you could make a switch of the status code checks.

It's minor code readability improvements, the code itself works as is.

Reply:
Thanks a lot already for the review, will try to implement your suggestions and request another review when I'm done, if that's fine with you 😅